Tests/TracingTests/SpanTests.swift

```diff
@@ -197,6 +197,103 @@ final class SpanTests: XCTestCase {
         }
         XCTAssertEqual(statusCode, 418)
     }
+
+    func testAttributesAsStringMetadata() {
+        var attributes = SpanAttributes()
+        attributes["http.status_code"] = 418
+        attributes["http.target"] = "target"
+
+        var metadata: [String: SwiftLogMetadataValue] = [:]
+        metadata[attributeKey: "http_target"] = attributes["http.target"]
+        #if swift(>=5.2)
+        metadata[attributeKey: "http_status_code"] = attributes.sampleHttp.statusCode
```

Comment:
This reads much better:

```swift
        metadata[attributeKey: "http_status_code"] = attributes.sampleHttp.statusCode
```

but we're back in the problem of one library having to depend on the other in order to provide the conversions. When we hit this earlier we thought that tracing depending on logging would not be too bad since it is the "heavier" of the two, so we could provide the conversions here...

Arguably the subscript is problematic here though, because of the differences in expressive power -- so we can't easily roundtrip (See the impls below). So maybe this should be:

```swift
metadata.setAttribute(attributes.sampleHttp.statusCode)
metadata.setAttribute("http_status_code", attributes.sampleHttp.statusCode)
```

?